### PR TITLE
Make input labels compliant with WCAG

### PR DESCRIPTION
**Pull request by jrpool**
Original version was not accessible for users of assistive technologies.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Labels are not programmatically associated with their form controls.
Issue Number: N/A

## What is the new behavior?
Labels are programmatically associated with their form controls, so assistive technologies will announce the labels when the controls are focused and the labels will extend the areas of clickability or touchability.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/examples/getting-started/src/app/cart/cart.component.html [hunk 1]
```diff
@@ -16,13 +16,17 @@ <h3>Cart</h3>
 <!-- #enddocregion checkout-form-1 -->
 
   <div>
-    <label>Name</label>
-    <input type="text" formControlName="name">
+    <label>
```
Comment (crisbeto) [inline aio/content/examples/getting-started/src/app/cart/cart.component.html]: I believe the correct thing to do here is to add an `id` to the `input` and to point the label's `for` attribute to the input's `id`.

**Review by Kuzcoo — COMMENTED** [review]
@jrpool : nice to see someone made a PR for that, I was going to correct it myself.
I think it is important to write accessible and semantic HTML in documentation, especially in "getting started" which is widely read.

I guess you should use "id" and "for" attributes as suggested by @crisbeto, in order to not break the initial style of the form.

**Review by brandonroberts — CHANGES_REQUESTED** [review]
Use `for` on the `label` to point to the input

**Comment by brandonroberts** [comment]
@jrpool also the commit message should be `docs: make input labels in getting started compliant with WCAG`

**Comment by jrpool** [comment]
I believe that the requested changes were completed in commit f00cc325e6ab1d245c7dfc8604d8cbe24da71c55. Is there more to do?

**Comment by brandonroberts** [comment]
@jrpool the changes LGTM, but you need to squash down to a single commit with `docs: change to explicit label type in getting-started form` as the message

**Comment by jrpool** [comment]
Squashed and remessaged. Please advise if more to do.

**Review by kapunahelewong — APPROVED** [review]
Thank you!

**Comment by jrpool** [comment]
Thanks! Why the notices about unmerged commits and CI failures?

**Comment by kapunahelewong** [comment]
@jrpool if memory serves me, the CI failures maybe had to do with the commit message format. They could also have had to do with codefresh flaking (or some other test) but it would have been nothing you did. Before merge, when you can see the tests, you can click on the details and scroll through to find the failures (or just do a find for "fail" or  "error") and they vary. Sometimes they are unrelated to a PR's changes. Since they're gone now I just have to guess.

Regarding the "unmerged commits" message, that one threw me at first too! 😁
Per @gkalpak, it's because of how we merge PRs using `rebase` workflow and GitHub can't tell a PR was merged into master. He shared two links to help me understand that you might also find interesting:

- Merging vs. Rebasing: https://www.atlassian.com/git/tutorials/merging-vs-rebasing
- How we merge PRs (so interesting!): https://github.com/angular/angular/blob/c845a7b887d1a85a89acfe5c62b8829079130af3/scripts/github/merge-pr#L201-L213

Also, in searching for an answer to your questions, I found a couple of helpful articles that helped me put some of the above into context. They are somewhat tangential but still ups one's git game 😊:

- https://help.github.com/en/articles/about-protected-branches
- https://mijingo.com/blog/creating-and-applying-patch-files-in-git

Thanks for asking these questions. Look at all the cool new stuff I learned in answering^^ 😲. Hope this helps!

**Comment by jrpool** [comment]
Thanks for your helpful reply, @kapunahelewong. I did check on the cause of the CI failure, and it was a bunch of 500 internal server errors.

As for git, it never stops puzzling me. I was ignorant of cherry picking and patch exporting until I looked at your references. These two techniques will be exactly what I need for a current project. Much appreciated.

**Comment by kapunahelewong** [comment]
Oh, maybe the 500 errors were related to the github outage today (maybe there were momentary, less noticeable server problems that popped up recently in these kinds of situations). I've noticed that sometimes tests just time out, too. That happens to me a fair amount.

I agree about the puzzle that is git! So glad the links helped. They helped me too. :)

**Pull request closed without merging**